show window when click system tray icon

kaichaosun opened the pull request:
* pop-up directly under system tray
* new AD4M brand style
* only show settings first and hide the rest in burger menu

<img width="705" alt="Screenshot 2022-08-23 at 11 46 47" src="https://user-images.githubusercontent.com/311749/186127860-7c108a30-f6f3-4e34-bf32-838867229753.png">

<img width="694" alt="Screenshot 2022-08-23 at 11 46 59" src="https://user-images.githubusercontent.com/311749/186127952-b6797a83-3d3f-4e67-b239-7621cf22ecd9.png">

kaichaosun committed:
system tray below window

kaichaosun committed:
Merge branch 'main' of github.com:perspect3vism/ad4min into cap-popup

kaichaosun committed:
make system tray window contains main section.

lucksus committed:
Move main window into system tray pop-up

lucksus committed:
Hide window on blur

lucksus committed:
Merge branch 'main' into cap-popup

lucksus committed:
Splashscreen AD4M branding

lucksus committed:
Adds new font Comfortaa and uses it in splashscreen

lucksus committed:
Deactivate event println!s

lucksus committed:
More AD4M branding

lucksus committed:
Set product name to AD4M

fayeed committed:
System tray window size changed

fayeed committed:
Updated styles

fayeed committed:
uncomment

fayeed committed:
Change navigation styles

lucksus committed:
cargo update

lucksus committed:
Fix circular starting of tauri binary by avoiding name collision

ad4m-host and ad4min would both build to an executable referenced to with "ad4m" at least inside the tauri virtual filesystem, which caused ad4min to start ad4min which first kills all "ad4m" processes... which lead to "it not working" in a weird way...

lucksus committed:
Merge branch 'main' into cap-popup

lucksus commented:
Ok, this is great and almost done. I just fixed the problem that made the app not work. Was a name collision of the binaries in the tauri fs. Had changed the product name (which is what will be shown in the macOS taskbar) to AD4M which also makes the produced binary be called "ad4m". But ad4m-host is also called "ad4m". That led to problems.

One thing I think is missing: a close button. Now there is no way to kill ad4min without using a system kill command or task manager...

fayeed committed:
Added a close button in tauri

fayeed committed:
Merge branch 'cap-popup' of github.com:perspect3vism/ad4min into cap-popup

fayeed commented:
@lucksus Added a close button in settings page

fayeed committed:
window now closes once agent unlocked

lucksus committed:
Bring back tray menu but only for Linux to prevent icon being hidden

lucksus committed:
warnings--

lucksus committed:
Remove unused SystemTrayWindow.tsx and route

lucksus committed:
Settings -> Main Settings with AD4M logo

lucksus committed:
Try to handle unsuccessful unlock

Window stays open, but error is not displayed. Seems like setState() in else-case in unlockAgent is not affecting the state

lucksus committed:
Move hasLoginError to AgentContext

jdeepee commented:
@lucksus this is working fine for me on Linux now, lets merge once we have the new ad4m-host version
